framework/encode/vulkan_handle_wrappers.h

```diff
@@ -517,25 +517,37 @@ struct CommandPoolWrapper : public HandleWrapper<VkCommandPool>
     bool           trim_command_pool{ false };
 };
 
-// For vkGetPhysicalDeviceSurfaceCapabilitiesKHR
 struct SurfaceCapabilities
 {
-    VkPhysicalDeviceSurfaceInfo2KHR surface_info;
-    std::unique_ptr<uint8_t[]>      surface_info_pnext_memory;
+    // For vkGetPhysicalDeviceSurfaceCapabilitiesKHR
+    VkSurfaceKHR             surface{ VK_NULL_HANDLE };
+    VkSurfaceCapabilitiesKHR surface_capabilities;
 
-    VkSurfaceCapabilities2KHR  surface_capabilities;
-    std::unique_ptr<uint8_t[]> surface_capabilities_pnext_memory;
+    // For vkGetPhysicalDeviceSurfaceCapabilities2KHR
+    bool use_surface_capabilities2{ false };
```

Comment:
Is it possible that we need to differentiate even between `2` and `2KHR`?

I was doing a similar change for `vkCreateRenderPass` and it made a difference whether the application was calling `2` and I latter called `2KHR` with getting a warning `Unsupported function vkCreateRenderPass2KHR was called, resulting in no-op behavior`.

Reply:
doh, you're right